dataclasses_avroschema/fields.py

```diff
@@ -407,6 +407,12 @@ def get_default_value(self):
 
 class LogicalTypeField(BaseField):
     def get_avro_type(self):
+        if self.default is not dataclasses.MISSING:
+            if self.default is not None:
+                return self.avro_type
```

Comment:
From my understating : 
* if the field declaration don't have any assignment (MISSING) then the assumption is that the value cant' be null => simple type
* if None is assigned as the default value, then obviously, it can be null => [null, type]
* if Any other value is assigned as the default value, then field shouldn't be null neither. The use-case where you re-assign null to a field that is non null be default is mainly uncommon => simple type. But as for the first case, it is an assumption there is no wrong choice. From my standpoint, when I assign a non null default value to a field, most of the time I consider it as not nullable

Reply:
Yes, you are right. I was confused with the unions.